test/views/ContactDetailsGuidanceViewSpec.scala

```diff
@@ -16,86 +16,37 @@
-
-  val SUT: ContactDetailsGuidanceView = app.injector.instanceOf[ContactDetailsGuidanceView]
-  given request: Request[?]           = FakeRequest()
-  given Messages                      = app.injector.instanceOf[MessagesApi].preferred(request)
+class ContactDetailsGuidanceViewSpec extends ViewSpecBase[ContactDetailsGuidanceView] {
 
   "ContactDetailsGuidanceView" - {
-    "must generate a view" - {
-      val doc = Jsoup.parse(SUT().toString)
-
-      "with the correct heading" in {
-        val mainContent = doc.getElementById("main-content")
-
-        val h1 = mainContent.getElementsByTag("h1")
-        h1.size() mustBe 1
-
-        h1.get(0).text() mustBe "We need contact details for this [company/group]"
-      }
-
-      "with the correct paragraphs" in {
-        val mainContent = doc.getElementById("main-content")
-
-        val paragraphs = mainContent.getElementsByTag("p")
-        paragraphs.size() mustBe 4
-        List.from(paragraphs.iterator().asScala).foreach(p => p.attr("class") mustBe "govuk-body")
-
-        paragraphs
-          .get(0)
-          .text() mustBe "Provide HMRC with contact details for the person or team responsible for this company or group."
-        paragraphs
-          .get(1)
-          .text() mustBe "We’ll use these details to:"
-        paragraphs
-          .get(2)
-          .text() mustBe "You could also include the Senior Accounting Officer’s contact details so they can stay informed."
-        paragraphs.get(3).text() mustBe "Is this page not working properly? (opens in new tab)"
-      }
-
-      "with the correct bullet points" in {
-        val mainContent = doc.getElementById("main-content")
-
-        val ul = mainContent.getElementsByTag("ul")
-        ul.size() mustBe 1
-        ul.attr("class") mustBe "govuk-list govuk-list--bullet"
-
-        val li = ul.get(0).getElementsByTag("li")
-        li.size() mustBe 2
-
-        li.get(0)
-          .text() mustBe "contact the right person if we have questions about the company’s tax accounting arrangements"
-        li.get(1).text() mustBe "send confirmation when the notification and certificate have been submitted"
-      }
-
-      "must have a continue button" in {
-        val mainContent = doc.getElementById("main-content")
-
-        mainContent.getElementById("submit").text() mustBe "Continue"
-      }
-
-      "must show a back link" in {
-        val backLink = doc.getElementsByClass("govuk-back-link")
-        backLink.size() mustBe 1
-      }
+    val doc = Jsoup.parse(SUT().toString)
+    mustHaveCorrectPageHeading(doc, pageHeading)
+    mustShowCorrectParagraphsWithCorrectContent(doc, totalParagraphsCount, paragraphsContent)
+    mustShowCorrectBulletPointsWithCorrectContent(doc, totalBulletPointsCount, bulletPointsContent)
+    mustHaveSubmitButton(doc, submitButtonContent)
+    mustShowABackLink(doc)
+    mustShowIsThisPageNotWorkingProperlyLink(doc)
```

Comment:
Can we add some line spacing here please

Also debating if we should start a naming convention of starting/ending with `Test` for the methods that would output a test in order to differentiate from the extension methods that act like a selector etc.

we can also consider making these extension methods as well

e.g.

```
doc.mustHaveCorrectPageHeadingTest(pageHeading)
or
doc.testMustHaveCorrectPageHeading(pageHeading)
or
doc.genTestMustHaveCorrectPageHeading(pageHeading)
etc.
```

what do you think?